python/ray/ml/config.py

```diff
@@ -105,6 +115,47 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         bundles = trainer_bundle + worker_bundles
         return PlacementGroupFactory(bundles, strategy=self.placement_strategy)
 
+    @classmethod
+    def validate_config(
```

Comment:
Where is this called?

Reply:
In a separate pr - https://github.com/ray-project/ray/pull/23850/

Reply:
Can we call this in the base Trainer class and also have a class attribute containing the allowed keys? Then each sub-trainer can override this class attribute. This will force all the Trainers to be explicit on which keys are allowed, and makes it easier to see this when reading the code/API docs.

Reply:
like the idea of defining allowed keys explicitly.